Display an info icon to notify users about a crashed tab

ayoy opened the pull request:
<!--
Note: This template is a reminder of our Engineering Expectations and Definition of Done. Remove sections that don't apply to your changes.

⚠️ If you're an external contributor, please file an issue before working on a PR. Discussing your changes beforehand will help ensure they align with our roadmap and that your time is well spent.
-->

Task/Issue URL: https://app.asana.com/1/137249556945/project/72649045549333/task/1209977265286175?focus=true

### Description

### Testing Steps
<!-- Assume the reviewer is unfamiliar with this part of the app -->
1.
2.

### Impact and Risks
<!-- 
What's the impact on users if something goes wrong?

High: Could affect user privacy, lose user data, break core functionality
Medium: Could disrupt specific features or user flows
Low: Minor visual changes, small bug fixes, improvement to existing features
None: Internal tooling, documentation
-->

#### What could go wrong?
<!-- Describe specific scenarios and how you've addressed them -->

### Quality Considerations
<!-- 
Focus on what matters for your changes:
- What edge cases exist?
- How does this affect performance?
- What monitoring have you added?
- What documentation needs updating?
- How does this impact privacy/security?
-->

### Notes to Reviewer
<!-- Anything specific you want reviewers to focus on -->

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

ayoy commented on macOS/DuckDuckGo/Tab/Model/TabCrashIndicatorModel.swift:
```diff
@@ -0,0 +1,57 @@
+
+/// This class manages the visibility of tab crash indicator.
+///
+final class TabCrashIndicatorModel: ObservableObject {
```
The tests for this class will be added soon

jotaemepereira reviewed (commented):
@ayoy I’ve started testing and I found the following problem with pinned tabs. The icon action doesn’t work all the times. It happens for both the crash and the audio button icons. I’m attaching a video.

To reproduce it I run the app, open youtube.com, I pin it, and then I either start a video and I cannot tap the audio icon or if I make it crash I cannot tap the crash information icon.



https://github.com/user-attachments/assets/c6e4ef96-eb61-4548-b3ab-79dbc3f341ab

ayoy commented:
@jotaemepereira thanks for pointing that out! I could not reproduce it but I suppose I know where it’s coming from.

I reworked audio button from a view + tap gesture recognizer into a plain style button (and added the crash button the same way). That buys us the “pressed” state for free, but also the pinned tab view is a button itself, so putting button inside a button bites us back here, I believe. I reverted the buttons to plain views + tap gesture recognizer and I hope it will fix the issue for you. Please give it another try and let me know. Thanks again!

jotaemepereira commented on macOS/DuckDuckGo/PinnedTabs/View/PinnedTabView.swift:
```diff
@@ -256,27 +258,72 @@ struct PinnedTabInnerView: View {
+        .popover(isPresented: $tabCrashIndicatorModel.isShowingPopover, arrowEdge: .bottom) {
+            PopoverMessageView(
+                viewModel: .init(
+                    title: "This tab has crashed",
```
I know that we are putting this behind a feature flag, but then we will launch to all users. Are you going to add translations in a following PR?

ayoy replied:
Yes, absolutely. I’ll add translations in the next PR, and also I’ll try to preemptively request translations based on Windows copy. I decided not to put these strings in UserText to limit noise related to untranslated UserText strings while this is in main waiting for translations.

jotaemepereira commented on macOS/DuckDuckGo/TabBar/View/TabBarViewController.swift:
```diff
@@ -1220,6 +1221,31 @@ extension TabBarViewController: TabBarViewItemDelegate {
+
+        DispatchQueue.main.async {
+            let viewController = PopoverMessageViewController(
+                title: "This tab has crashed",
```
Same question as the last one, are we going to add translations for these strings?

jotaemepereira commented on macOS/UnitTests/Tab/Model/TabCrashIndicatorModelTests.swift:
```diff
@@ -0,0 +1,107 @@
+import XCTest
+@testable import DuckDuckGo_Privacy_Browser
+
+final class TabCrashIndicatorModelTests: XCTestCase {
```
👏🏼

jotaemepereira reviewed (approved):
This looks great @ayoy. Great job on all tests added 👏🏼
